### PR TITLE
[circle-eval-diff] Use CircleEvalDiff in driver

### PR DESCRIPTION
**Pull request by jinevening**
This uses CircleEvalDiff in driver.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/8634
Draft PR: https://github.com/Samsung/ONE/pull/8741

### CODE REVIEW

**Comment by jinevening** [comment]
To reviewers:

CircleEvalDiff.h is moved to `include` directory for being used by driver.

**Review by seanshpark — DISMISSED** [review]
LGTM

File: compiler/circle-eval-diff/driver/Driver.cpp [hunk 1]
```diff
@@ -14,12 +14,38 @@
...
 namespace
 {
 
+std::string to_lower_case(std::string s)
+{
+  std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) { return std::tolower(c); });
+  return s;
+}
+
+Metric str_to_metric(const std::string &str)
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/driver/Driver.cpp:33]: How about rename it to `to_metric` like other functions? `str` keyword in function name seems not needed.

File: compiler/circle-eval-diff/driver/Driver.cpp [hunk 2]
```diff
@@ -14,12 +14,38 @@
...
+{
+  if (to_lower_case(str).compare("mae") == 0)
+    return Metric::MAE;
+
+  throw std::runtime_error("Unsupported metric.");
+}
+
+InputFormat str_to_input_format(const std::string &str)
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/driver/Driver.cpp:41]: ditto

File: compiler/circle-eval-diff/driver/Driver.cpp [hunk 3]
```diff
@@ -87,7 +113,45 @@ int entry(const int argc, char **argv)
...
   }
 
-  // TODO Run CircleEvalDiff
+  const auto first_model_path = arser.get<std::string>("--first_model");
+  const auto second_model_path = arser.get<std::string>("--second_model");
+
+  // Default values
+  std::string first_input_data_path;
+  std::string second_input_data_path;
+  std::string metric("MAE");
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/driver/Driver.cpp:122]: `arser` has `default_value` method:) https://github.com/Samsung/ONE/blob/36872b7bef4eabbbe0457c5a716824a06c85f809/compiler/arser/tests/arser.test.cpp#L267-L271

File: compiler/circle-eval-diff/driver/Driver.cpp [hunk 4]
```diff
@@ -87,7 +113,45 @@ int entry(const int argc, char **argv)
...
+  std::string input_data_format("h5");
+
+  if (arser["--first_input_data"])
+    first_input_data_path = arser.get<std::string>("--first_input_data");
+
+  if (arser["--second_input_data"])
+    second_input_data_path = arser.get<std::string>("--second_input_data");
+
+  if ((not first_input_data_path.empty() and second_input_data_path.empty()) or
```
[the remark this answers is not in the document]
Comment (mhs4670go) [reply compiler/circle-eval-diff/driver/Driver.cpp:131]: Then, boolean value can represent an exclusive OR as `value1 != value2`.

File: compiler/circle-eval-diff/driver/Driver.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/circle-eval-diff/driver/Driver.cpp]: Ah, I made this draft before hearing your naming seminar and forgot to update it.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**